spec/unit/pdk/control_repo_spec.rb

```diff
@@ -0,0 +1,112 @@
+require 'spec_helper'
+require 'pdk/util'
+
+describe PDK::ControlRepo do
+  describe '.control_repo_root' do
+    [
+      { name: 'with strict_check set to false', strict_check: false },
+      { name: 'with strict_check set to true but is not a Bolt project dir', strict_check: true },
+    ].each do |testcase|
```

Comment:
Is there a reason to not just do this with shared examples and two explicit `context`s? Dynamically generated specs can be a pain to debug.

Reply:
Fixed